pkg/kptfile/kptfileutil/util.go

```diff
@@ -15,17 +15,38 @@
 package kptfileutil
 
 import (
+	"bytes"
 	"fmt"
 	"io/ioutil"
 	"os"
 	"path/filepath"
 	"strings"
 
+	"github.com/GoogleContainerTools/kpt/internal/util/git"
 	"github.com/GoogleContainerTools/kpt/pkg/kptfile"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"sigs.k8s.io/kustomize/kyaml/errors"
 	"sigs.k8s.io/kustomize/kyaml/yaml"
 )
 
+// UnknownKptfileVersionError is returned when kpt encounters a Kptfile
+// that uses an unsupported version of the Kptfile resource.
+type UnknownKptfileVersionError struct {
+	PkgPath  string
+	RepoSpec *git.RepoSpec
+	Version  string
+}
+
+func (e *UnknownKptfileVersionError) Error() string {
+	var source string
+	if e.RepoSpec != nil {
+		source = e.RepoSpec.RepoRef()
+	} else {
+		source = e.PkgPath
+	}
+	return fmt.Sprintf("package at %q is using an unknown version of the Kptfile schema. It might be for a newer version of kpt. Please try updating kpt.", source)
```

Comment:
1. We should include the version (when available) in the message. Actually semver pkgs helps in comparing versions, so saying something like `pkg at "x" is using newer version "v1.2.2" of Kptfile. Please try updating kpt.` will read much better. 
2. I think we can remove the `schema` word.

I realized we will need something similar for kpt CLI 1.0 encountering older versions of `Kptfile` that are incompatible. There we need to point them to migration guide.

Reply:
Updated the PR to print the version as part of the message. Also update the wording of the error message, but I think we should keep the `schema` word.

PR for handling kpt 1.0 encountering older versions of the Kptfile schema is here: https://github.com/GoogleContainerTools/kpt/pull/2161

Reply:
Ack.